GameServer/Logic/Packets/Packets.cs

```diff
@@ -43,7 +43,7 @@ public class BasePacket : Packet
     {
         public BasePacket(PacketCmdS2C cmd = PacketCmdS2C.PKT_S2C_KeyCheck, uint netId = 0) : base(cmd)
         {
-            buffer.Write((uint)netId);
+            buffer.Write(netId);
```

Comment:
I think the reason these are cast is to better keep track of what kind of data is written to the buffer / makes it easier to interpret. So I'd vote for not (yet at least) changing the casts in buffer writes, or at least adding the structure as a comment / docstring. This goes for all packet buffer writes.